src/lib/lib_cron.py

```diff
@@ -63,11 +63,18 @@ def install_cronjob(self):
 
             logrotate_unit = hookenv.local_unit()
             python_venv_path = os.getcwd().replace("charm", "") + ".venv/bin/python3"
+            if hookenv.has_juju_version("3.0"):
+                juju_exec = "juju-exec"
+            else:
+                juju_exec = "juju-run"
             # upgrade to template if logic increases
             cron_job = """#!/bin/bash
-/usr/bin/sudo /usr/bin/juju-run {} "{} {}"
+/usr/bin/sudo /usr/bin/{juju_exec} {logrotate_unit} "{python_venv_path} {cronjob_path}"
```

Comment:
I think we should use [shlex](https://docs.python.org/3/library/shlex.html) to be more safe here. WDYT?

Reply:
Nothing is coming from config option? ok resolving.